plutus-ir/test/Spec.hs

```diff
@@ -24,20 +23,31 @@ import           Language.PlutusCore.StdLib.Type
 
 import           Test.Tasty
 
+import           Control.Exception
 import           Control.Monad
 import           Control.Monad.Except
+import           Control.Monad.Reader
 
 main :: IO ()
 main = defaultMain $ runTestNestedIn ["test"] tests
 
+-- for throwing away the annotation
+instance GetProgram (PLC.Term TyName Name (Provenance ())) where
+    getProgram = getProgram . void
+
+-- normal compilation with typechecking
 instance GetProgram (Quote (Term TyName Name ())) where
-    getProgram = trivialProgram . compileAndMaybeTypecheck False
+    getProgram = getProgram . compileAndMaybeTypecheck True
+
+-- convenience so we can also compile without typechecking
+instance (Exception e, GetProgram a) => GetProgram (Except e a) where
+    getProgram = either throw id . runExcept . fmap getProgram
```

Comment:
It looks weird to me, how many distinct things `getProgram` allows to do.

Reply:
I have a couple of ideas, but I'll try them out in a follow up, if that's okay.

Reply:
Sure.